stablehlo/dialect/TypeInference.cpp

```diff
@@ -1114,20 +1124,26 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
   SmallVector<int64_t, 4> shape;
   shape.reserve(rank);
   SmallVector<int64_t> resultBounds;
+
+  auto inferSliceDim = [&](int64_t inputDim, int64_t start, int64_t end,
+                           int64_t stride) {
+    return inputDim == ShapedType::kDynamicSize
+               ? ShapedType::kDynamicSize
+               : static_cast<int64_t>(llvm::divideCeil(end - start, stride));
+  };
+
   for (int64_t i = 0, e = rank; i != e; i++) {
-    if (!inputBounds.empty())
-      resultBounds.push_back(
-          inferSliceDim(inputBounds[i], start[i], limit[i], strideVals[i]));
     if (hlo::isDynamicDimSize(rankedTy.getDimSize(i))) {
       shape.push_back(ShapedType::kDynamicSize);
-      continue;
+      if (inputBounds.empty()) continue;  // dynamic dim without bounds
     }
     // P3.
     if (start[i] < 0)
```

Comment:
Good find! It is always better to verify these attributes even if they are not used?

Also, it seems slice op change is completely unrelated to pad inference.  If that is correct, let's spin up another PR to keep the changes focused?

https://google.github.io/eng-practices/review/developer/small-cls.html

Reply:
Small PRs: totally agreed. Will send the other one soon later.
"verify these attributes even if they are not used": No, I do not intent to restrict the verifications in any way, with the concern that it may break MHLO further. I mean, for "unbounded dynamic" case, we did not check the attrs, and I keep it as-is in this PR (simply continue the loop). But for bounded dynamic case, if the attrs are used to calculate the result_bound, then I suggest to check them, like we check them for "static dim case.

Reply:
Thanks! yes, the other PR has checks in the front and seems safer.